adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -61,6 +64,20 @@ public void onChange(BaseRealm results) {
         }
     }
 
+    public RealmBaseAdapter(@Nullable OrderedRealmCollection<T> data) {
+        this.adapterData = data;
```

Comment:
We should also check if the data is managed otherwise adding listeners might crash. RealmCollection has a `isManaged()` method now, but that is only available from 2.x.